src/main/scala/com/fulcrumgenomics/bam/ErrorRateByReadPosition.scala

```diff
@@ -0,0 +1,280 @@
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ *
+ */
+
+package com.fulcrumgenomics.bam
+
+import com.fulcrumgenomics.cmdline.{ClpGroups, FgBioTool}
+import com.fulcrumgenomics.util.{Metric, NumericCounter, ProgressLogger}
+import dagr.commons.CommonsDef._
+import dagr.commons.io.{Io, PathUtil}
+import dagr.commons.util.LazyLogging
+import dagr.sopt._
+import htsjdk.samtools.SAMFileHeader.SortOrder
+import htsjdk.samtools.reference.ReferenceSequenceFileFactory
+import htsjdk.samtools.util.SequenceUtil
+import htsjdk.samtools.util.SequenceUtil.{isValidBase => validBase}
+import htsjdk.samtools.{CigarOperator, SAMRecord, SamReaderFactory}
+
+@clp(description =
+  """
+    |Calculates the error rate by read position on mapped BAMs.
+    |
+    |If a reference is given, the sort order must be coordinate.  If no reference is given, the MD tag must be present
+    |for every mapped read and computed correctly, and no sort order is required.
+    |
+    |The output metrics text file will have the extension 'error_rate_by_read_position.txt'.  The output file will
+    |contain a row per read and per position in that read.  The 'read_number' will be 0 for fragments, 1 for the first
+    |end of a pair, and 2 for the second end of a pair. Both the error rate and number of bases observed will be
+    |provided.  Soft-clipped bases, and unmapped, secondary, supplementary, and vendor failing reads are ignored.
+    |Optionally, duplicates are ignored.
+  """,
+  group = ClpGroups.SamOrBam)
+class ErrorRateByReadPosition
+( @arg(flag="i", doc="Input BAM file.") val input: PathToBam,
+  @arg(flag="o", doc="Output metrics prefix. If not given, will use the input BAM basename as a prefix.")
+  val output: Option[PathPrefix] = None,
+  @arg(flag="r", doc="Reference fasta. If not given, will use the MD tag.") val ref: Option[PathToFasta] = None,
+  @arg(flag="d", doc="Include duplicate reads, otherwise ignore.") val duplicates: Boolean = false,
+  @arg(flag="I", doc="Count insertions as errors, otherwise as matches.") val includeInsertions: Boolean = false,
+  @arg(flag="m", doc="The minimum mapping quality for a read to be included.") val minMappingQuality: Int = 0,
```

Comment:
I think it probably makes sense to have a higher default for MQ, since you want to make sure your reads are mapped to the right place and the errors you are seeing aren't mapping artifacts.